src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -1216,6 +1218,127 @@ class CopyOptionsSegment(BaseSegment):
     )
 
 
+@snowflake_dialect.segment(replace=True)
+class CreateSchemaStatementSegment(BaseSegment):
+    """A `CREATE SCHEMA` statement.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-schema.html
+    """
+
+    type = "create_schema_statement"
+    match_grammar = Sequence(
+        "CREATE",
+        Ref("OrReplaceGrammar", optional=True),
+        Ref("TemporaryTransientGrammar", optional=True),
+        "SCHEMA",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("SchemaReferenceSegment"),
+        Sequence("WITH", "MANAGED", "ACCESS", optional=True),
```

Comment:
Everything here and below in not order dependent. Can you wrap this and below in AnyNumberOf to allow for this?

Reply:
I am not sure I understand what you mean by that, because the order is important when running these statements.

Running `create schema foo DATA_RETENTION_TIME_IN_DAYS = 1 with managed access;` will return this error `SQL compilation error: syntax error line 1 at position 55 unexpected 'managed'.` And running `create schema foo tag (zort='bar') MAX_DATA_EXTENSION_TIME_IN_DAYS = 1;` will return `SQL compilation error: syntax error line 1 at position 35 unexpected 'MAX_DATA_EXTENSION_TIME_IN_DAYS'.`. Finally `create schema foo MAX_DATA_EXTENSION_TIME_IN_DAYS = 1 with tag (zort='bar') comment = 'foo';` will return `SQL compilation error: syntax error line 1 at position 76 unexpected 'comment'.`.

However, `DATA_RETENTION_TIME_IN_DAYS`, `MAX_DATA_EXTENSION_TIME_IN_DAYS`, `DEFAULT_DDL_COLLATION` and `COMMENT` can indeed be placed in arbitrary order, as allowed by  `SchemaObjectParamsSegment`.

Did I misunderstand what you were suggesting?

Reply:
Ah okay - not what I expected